libfwupdplugin/fu-device.c

```diff
@@ -2497,6 +2498,72 @@ fu_device_add_guid_full(FuDevice *self, const gchar *guid, FuDeviceInstanceFlags
 	fu_device_add_guid_safe(self, guid, flags);
 }
 
+/**
+ * fu_device_has_counterpart_guid:
+ * @self: a #FuDevice
+ * @guid: a GUID, e.g. `2082b5e0-7a64-478a-b1b2-e3404fab6dad`
+ *
+ * Finds out if the device has a specific counterpart GUID.
+ *
+ * Returns: %TRUE if the counterpart GUID is found
+ *
+ * Since: 1.9.21
+ **/
+gboolean
+fu_device_has_counterpart_guid(FuDevice *self, const gchar *guid)
+{
+	FuDevicePrivate *priv = GET_PRIVATE(self);
+
+	g_return_val_if_fail(FU_IS_DEVICE(self), FALSE);
+	g_return_val_if_fail(guid != NULL, FALSE);
+
+	/* convert and try again */
+	if (!fwupd_guid_is_valid(guid)) {
+		g_autofree gchar *tmp = fwupd_guid_hash_string(guid);
+		return fu_device_has_counterpart_guid(self, tmp);
+	}
+
+	/* any defined? */
```

Comment:
I think this should be first check of the function. No use recursing with the above call if there are none to start with.